staging/src/k8s.io/apiserver/pkg/server/options/authorization.go

```diff
@@ -151,9 +151,12 @@ func (s *DelegatingAuthorizationOptions) getClient() (kubernetes.Interface, erro
 		clientConfig, err = loader.ClientConfig()
 	} else {
 		// without the remote kubeconfig file, try to use the in-cluster config.  Most addon API servers will
-		// use this path
+		// use this path. If it is optional, ignore errors.
 		clientConfig, err = rest.InClusterConfig()
-		if err == rest.ErrNotInCluster && s.RemoteKubeConfigFileOptional {
+		if err != nil && s.RemoteKubeConfigFileOptional {
+			if err != rest.ErrNotInCluster {
```

Comment:
then i wonder if this check could be gone?

Reply:
No. The warning below is only for in-cluster usage. Even in cluster, there might be no service account and therefore no token (in static pods). So the warning will make that clear. But it would be misleading for the out-of-cluster use-case.